aws/ecs/packer/to-upload/peers.sh

```diff
@@ -2,29 +2,221 @@
+      >&2 echo "${FUNCNAME[0]}: Could not find aws-cli"
+      exit 2
+    fi
+    __awscli="${system_awscli} --region ${__region} "
+  fi
+}
+
+get_instance_autoscaling_group() {
+  if ! check_data "${__instance_autoscaling_group}"; then
+    get_awscli
+    get_instance_id
+    local __autoscaling_group_filter="AutoScalingInstances[?InstanceId == '${__instance_id}'].AutoScalingGroupName"
+    __instance_autoscaling_group="$(${__awscli} autoscaling describe-auto-scaling-instances --query "${__autoscaling_group_filter}" --output text)"
+    if ! check_data "${__instance_autoscaling_group}"; then
+      >&2 echo "${FUNCNAME[0]}: WARNING Instance does not belong to an autoscaling group"
+      return 1
+    fi
+  fi
+}
+
+get_autoscaling_peer_instance_ids() {
+  if ! check_data "${__peer_autoscaling_instance_ids}"; then
+    get_awscli
+    if get_instance_autoscaling_group; then
+      get_instance_id
+      local __peer_instances_filter="AutoScalingInstances[?AutoScalingGroupName == '${__instance_autoscaling_group}']|[?InstanceId != '${__instance_id}'].InstanceId"
+      __peer_autoscaling_instance_ids="$(${__awscli} autoscaling describe-auto-scaling-instances --query "${__peer_instances_filter}" --output text)"
+      check_exit $?
+      if ! check_data "${__peer_autoscaling_instance_ids}"; then
+        >&2 echo "${FUNCNAME[0]}: WARNING autoscaling group has no other instances"
+        return 1
+      fi
+    else
+      return 1
+    fi
+  fi
+}
 
-peer_ips=$($aws ec2 describe-instances --instance-ids $peer_instances | \
-	jq -r '.Reservations | .[] | .Instances | .[] | .NetworkInterfaces | .[] | .PrivateIpAddress')
+get_list_of_active_spot_fleets() {
+  if ! check_data "${__list_of_active_spot_fleets}" ;then
+    get_awscli
+    local __spot_fleet_active_filter="SpotFleetRequestConfigs[?SpotFleetRequestState == 'active'].SpotFleetRequestId"
+    __list_of_active_spot_fleets="$(${__awscli} ec2 describe-spot-fleet-requests --query "${__spot_fleet_active_filter}" --output text)"
+    check_exit $?
+    if ! check_data "${__list_of_active_spot_fleets}"; then
+      >&2 echo "${FUNCNAME[0]}: WARNING There are no active spot fleets"
+      return 1
+    fi
+  fi
+}
+
+get_spot_fleet_request_id() {
+  if ! check_data "${__spot_fleet_request_id}"; then
+    get_awscli
+    get_instance_id
+    if get_list_of_active_spot_fleets; then
+      local __spot_fleet_instance_filter="ActiveInstances[?InstanceId == '${__instance_id}'].InstanceId"
+      for spot_fleet_id in "${__list_of_active_spot_fleets}"; do
+        local __possible_spot_fleet="$(${__awscli} ec2 describe-spot-fleet-instances --spot-fleet-request-id "${spot_fleet_id}" --query "${__spot_fleet_instance_filter}" --output text)"
+        if [ "${__possible_spot_fleet}" == "${__instance_id}" ]; then
+          __spot_fleet_request_id="${spot_fleet_id}"
+          break
+        fi
+      done
+      if ! check_data "${__spot_fleet_request_id}"; then
```

Comment:
I don't see a problem here.  What check are you referring to that needs refactoring?